jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala

```diff
@@ -205,6 +208,43 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
     }
   }
 
+  /**
+   * Find all BasicScanExecTransformers in one WholeStageTransformerExec
+   */
+  def checkBatchScanExecTransformerChildren(): Seq[BasicScanExecTransformer] = {
+    val basicScanExecTransformers = new mutable.ListBuffer[BasicScanExecTransformer]()
+
+    def transformChildren(plan: SparkPlan,
+                          basicScanExecTransformers: mutable.ListBuffer[BasicScanExecTransformer]
+                         ): Unit = {
+      if (plan != null && plan.isInstanceOf[TransformSupport]) {
+        if (plan.isInstanceOf[BasicScanExecTransformer]) {
+          basicScanExecTransformers.append(plan.asInstanceOf[BasicScanExecTransformer])
+        }
+        // according to the substrait plan order
+        if (plan.isInstanceOf[ShuffledHashJoinExecTransformer]) {
```

Comment:
Is the special handling for other join also needed? Maybe change to HashJoinLikeExecTransformer and also consider SortMergeJoinExecTransformer.

Reply:
Good suggestion, fixed.